### ISSUE TITLE
Metrics: if oauth relier doesn't provide entrypoint, set entrypoint as the client_id

### ISSUE THREAD

**Issue by vladikoff** [mozilla/fxa-content-server#6757]
if oauth relier doesn't provide entrypoint, set entrypoint as the client_id

@davismtl @irrationalagent what do you think about this idea?

See this chart with entrypoints and client ids: https://analytics.amplitude.com/mozilla-corp/chart/2ngrhz1/edit/nkdi8ub

**Comment by philbooth** [comment]
I don't get the point of this. We already include the client id and service name as event properties. What's the properties of including one of them again under a different property name?

**Comment by vladikoff** [comment]
> What's the point of including one of them again under a different property name?

We are trying to figure out how to minimize or eliminate empty entrypoints

**Comment by vladikoff** [comment]
@philbooth there is a good comment here: https://github.com/mozilla/fxa-content-server/issues/6756#issuecomment-447414679

**Comment by shane-tomlinson** [comment]
> I think we should in parallel identify where these empty entrypoints are coming from and have them fixed.

I started looking on Friday, in amplitude search for logins & creates where entrypoint = none and group by client_id. I'll generate a list today.

**Comment by shane-tomlinson** [comment]
> I started looking on Friday, in amplitude search for logins & creates where entrypoint = none and group by client_id

[here we go](https://analytics.amplitude.com/mozilla-corp/chart/y8x5v48). I'll put names to client_ids.

**Comment by shane-tomlinson** [comment]
>  I'll put names to client_ids.

I have emailed the team the list of reliers.

**Comment by irrationalagent** [comment]
There is a good reason to do this: since the oauth client id parameter is passed only as an event property to amplitude, they can't be used easily to segment users for (e.g.) retention purposes. So I'm 100% to do this!

**Comment by irrationalagent** [comment]
copypasta from another issue,

I just posted this in IRC, something I realized, maybe I need to file a separate issue.

I think I've seen some cases where entrypoint is defined (for example) in the first few steps of a funnel (view events) but not for later steps. This causes entrypoint to get set as a user property which persists, allowing us to track the origin of a user's registration. I wonder then, if we eliminate all null values by filling in with service/client_id, then will subsequent funnel steps now clobber the user property undesirably?

see [this for example](https://analytics.amplitude.com/mozilla-corp/chart/new/kp0lr36) the about:welcome (activity-stream-firstrun) entrypoint is defined for the email-first view but not subsequent steps. If a user completes registration through that entrypoint, then they still receive the correct entrypoint as a user property, since all subsequent steps did not have it set.

If we start filling in the null values for the post-email-first steps with sync or something else, I think they would then clobber activity-stream-firstrun ?

this seems to be especially true for self-hosted email forms, which are now becoming the norm.

**Comment by davismtl** [comment]
@irrationalagent can you run this by Danny from Amplitude? He can help confirm or deny our assumptions.

**Comment by irrationalagent** [comment]
In talking with the folks in amplitude, they clarified that only the values of the user properties set at the time of the first step in the funnel chart matter for the calculation of the results, so making this change won't break our funnels.

However, if we start setting empty `entrypoint` values to equal `client_id`, then we will have to keep in mind that in some cases intermediate funnel steps (such as `engage` and `submit`) will have different values than the initial step. For most purposes I think this will be ok, but it could lead to some confusion if, e.g. we ever needed to explicitly count `submit` steps per entrypoint for some reason.

So I'm going to tentatively say that we should move forward with this.  I'm going to post in https://github.com/mozilla/fxa-content-server/issues/6756 about the cases where `entrypoint` parameters are not being carried forward to the login/reg content server events

**Comment by shane-tomlinson** [comment]
In the process of reverting the change that closed this, re-opening.
